sklearn/_build_utils/__init__.py

```diff
@@ -62,6 +65,8 @@ def build_from_c_and_cpp_files(extensions):
 
 def maybe_cythonize_extensions(top_path, config):
     """Tweaks for building extensions between release and development mode."""
+    with_openmp = check_openmp_support()
```

Comment:
Put this below right before the call to `cythonize`?

Reply:
I can't because `check_openmp_support` does a check which needs to be done before any build command.

Reply:
I don't see any build command here?

Reply:
`build_from_c_and_cpp_files(config.ext_modules)`

Reply:
Sorry if I'm missing something obvious, but this doesn't build anything

Reply:
you're right it does not compile. neither does the following. But when you run `python setup.py build_ext -i`, this function will be called, cythonization will be done if necessary, and then the compiler will compile the c files. So the check of openmp support has to be done whether cythonization happens or not.

Reply:
For the C files to exist, cythonization has to happen at some prior point, so I don't see why we need this here.

Reply:
the C compilation involves openmp as well, and if you set the `-fopenmp` flag but the compiler does not support openmp, the build will fail

Reply:
but the build would have failed **before** already, during the cythonization

Reply:
When we make a release (or for the nightly builds), we build the wheels, but we also provide the sources for when no wheel matches the user's system. In the second case, we only provide the already C generated files. So only the C compilation happens on the user's system.

Reply:
Ok, I was assuming that we would simply ship the .pyx files and require the builders to use Cython.

It makes sense then, thanks